### PR TITLE
ConnectRequestFragment

### PR DESCRIPTION
**Pull request by makingthematrix**
#### APK
[Download build #812](http://10.10.124.11:8080/job/Pull%20Request%20Builder/812/artifact/build/artifact/wire-dev-PR2531-812.apk)
[Download build #837](http://10.10.124.11:8080/job/Pull%20Request%20Builder/837/artifact/build/artifact/wire-dev-PR2531-837.apk)
[Download build #903](http://10.10.124.11:8080/job/Pull%20Request%20Builder/903/artifact/build/artifact/wire-dev-PR2531-903.apk)
[Download build #907](http://10.10.124.11:8080/job/Pull%20Request%20Builder/907/artifact/build/artifact/wire-dev-PR2531-907.apk)
[Download build #930](http://10.10.124.11:8080/job/Pull%20Request%20Builder/930/artifact/build/artifact/wire-dev-PR2531-930.apk)
[Download build #950](http://10.10.124.11:8080/job/Pull%20Request%20Builder/950/artifact/build/artifact/wire-dev-PR2531-950.apk)

### CODE REVIEW

File: app/src/main/res/layout/fragment_participants_connect.xml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:layout_weight="1"
+        android:layout_marginTop="@dimen/wire__padding__regular"
+        >
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/not_tabbed_recycler_view"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            />
```
Comment (BradleyWire) [inline app/src/main/res/layout/fragment_participants_connect.xml]: formatting

File: app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala [hunk 2]
```diff
@@ -298,7 +298,14 @@ class ConversationListManagerFragment extends Fragment
               navController.setLeftPage(Page.SEND_CONNECT_REQUEST, Tag)
             }
 
-          case PENDING_FROM_OTHER | PENDING_FROM_USER | IGNORED =>
+          case PENDING_FROM_OTHER  =>
+            show(
+              ConnectRequestFragment.newInstance(userId, userRequester),
+              ConnectRequestFragment.Tag
+            )
+            navController.setLeftPage(Page.PENDING_CONNECT_REQUEST, Tag)
```
Comment (BradleyWire) [inline app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala]: Is this Page reference correct? It doesn't look like it's pending

File: app/src/main/scala/com/waz/zclient/participants/fragments/ParticipantFragment.scala [hunk 3]
```diff
@@ -48,9 +47,7 @@ import scala.concurrent.Future
 
 class ParticipantFragment extends ManagerFragment
   with ConversationScreenControllerObserver
-  with SendConnectRequestFragment.Container
-  with BlockedUserProfileFragment.Container
-  with PendingConnectRequestFragment.Container {
```
[the remark this answers is not in the document]
Comment (makingthematrix) [reply app/src/main/scala/com/waz/zclient/participants/fragments/ParticipantFragment.scala]: They're not needed anymore. They were used mainly for making other fragments react to certain operation inside these three (mainly tapping the back button). The newer solution is to contain all logic handling that inside the given fragment.

File: app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala (hunk 2 above)
Comment (makingthematrix) [reply app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala]: I think in this case it doesn't matter - it's here just to make the code consistent with other options.

**Pull request merged**